### PR TITLE
feature: Add support for deletion_protection_enabled attribute for DynamoDB

### PR DESCRIPTION
**Pull request by Hamza-Aziz**
## what

This PR add support for the attribute `deletion_protection_enabled` in the DynamoDB ressource
## why

To address an issue https://github.com/cloudposse/terraform-aws-tfstate-backend/issues/143
To present or force DynamoDB table deletion

## references
To address an issue https://github.com/cloudposse/terraform-aws-tfstate-backend/issues/143
https://aws.amazon.com/about-aws/whats-new/2023/03/amazon-dynamodb-table-deletion-protection/

### CODE REVIEW

**Comment by hans-d** [comment]
@Hamza-Aziz thanks!
Can you follow up on https://github.com/cloudposse/terraform-aws-tfstate-backend/actions/runs/8110694935/job/22170394750?pr=156 - at the end is states what needs to be done.

**Comment by Hamza-Aziz** [comment]
@hans-d thanks for follow up
`make init` works but when I run `make readme` it gives:
`make: *** No rule to make target readme.  Stop`

**Comment by hans-d** [comment]
>  `make: *** No rule to make target readme. Stop`

strange, will have look at that. so far havent seen any such issues.

File: variables.tf [hunk 1]
```diff
@@ -34,6 +34,13 @@ variable "force_destroy" {
   default     = false
 }
 
+variable "deletion_protection_enabled" {
+  type        = bool
+  description = "A boolean that enables deletion protection for DynamoDB table"
+  default     = true
```
Comment (hans-d) [inline variables.tf:40]: Enabling by default breaks backward compatibility. Also needs the currents tests to be updated.

**Comment by hans-d** [comment]
> > `make: *** No rule to make target readme. Stop`
> 
> strange, will have look at that. so far havent seen any such issues.

It's working for me when I check out your branch. Normally it should be pulling in the build-harness and doing its magic...
```
❯ make readme
* Installing packages master...
Cloning into '/Users/me/github/terraform-aws-tfstate-backend/build-harness/vendor/packages'...
```

**Comment by Gowiem** [comment]
@Hamza-Aziz before running `make readme`, you do need to run `make init` to pull the build-harness, which I don't think was made clear. 

@hans-d still something up with the readme... Maybe the lockfile issue? Also, for your future self, I have the following GH "Saved reply" for pinging people to request readme changes: 

> Unfortunately, we need to run some automation to pass our tests. Mind doing the following locally, adding + committing the result, and pushing to your branch?
> 
> ```
> make init
> make readme
> ```
> 
> Thanks!

**Comment by Hamza-Aziz** [comment]
@Gowiem Thank you for your follow-up. As mentioned in my previous comment, I began by running `make init`, and it indeed pulled the build harness. However, when I proceeded to run `make readme`, I encountered the following error:

> make: *** No rule to make target 'readme'. Stop.


Do I need to change the directory? I'm executing the commands at the same level as the 'Makefile.'"

**Comment by hans-d** [comment]
@Hamza-Aziz Make file should indeed be in the repository root.
Currently tried it again (as some upstream fix is now in that should solve one other issue we were having).

Re why it doesnt work for you, that is a bit vague. Can you share info about operating system and such?

**Review by hans-d — DISMISSED** [review]

File: variables.tf (hunk 1 above)
Comment (hans-d) [reply variables.tf:40]: still applies

Comment (hans-d) [reply variables.tf:40]: @Hamza-Aziz can you change this? Than I'm happy to merge

**Comment by Hamza-Aziz** [comment]
@hans-d done

File: variables.tf (hunk 1 above)
Comment (Hamza-Aziz) [reply variables.tf]: @hans-d done

**Review by hans-d — APPROVED** [review]

**Comment by hans-d** [comment]
@Hamza-Aziz thanks for the ride!

**Comment by Hamza-Aziz** [comment]
@hans-d You're welcome! Thanks for your assistance.

**Pull request merged**